<launch>
    <!-- play bag -->
    <include file="$(find sim2real)/launch/bag_play.launch">
        <arg name="play_bagfile" value="$(find sim2real)/rosbag/0424/remote_control_image.bag"/>
        <arg name="save_bagfile" value="$(find sim2real)/rosbag/0424/proc.bag"/>
        <arg name="record"  value="false"/>
        <arg name="rviz"   value="true"/>
        <arg name="rviz_config" value="$(find sim2real)/rviz/carto.rviz"/>
    </include>

    <!-- PointCloud to LaserScan conversion -->
    <node name="pointcloud_to_laserscan" pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node">
        <!-- Parameters determined by livox radar  -->
        <remap from="cloud_in" to="/livox/lidar_inv"/>
        <param name="min_height" value=""/>   
        <param name="max_height" value="0.5"/> 
        <param name="angle_min" value="-3.14"/> 
        <param name="angle_max" value="3.14"/> 
        <param name="range_min" value="0.01"/> 
        <param name="range_max" value="8"/> 
        <param name="target_frame" value="livox_frame_inv"/>    
        <!-- Parameters  to be adjusted  -->
        <param name="angle_increment" value="0.004"/> 
        <param name="queue_size" value="10"/> 
        <param name="scan_time" value=""/> 
        <param name="transform_tolerance" value=""/>
        <param name="use_inf" value="true"/>
    </node>

    <!-- Cartographer 2D mapping -->
    <node name="cartographer_node" pkg="cartographer_ros"
        type="cartographer_node" args="
            -configuration_directory $(find sim2real)/config/carto
            -configuration_basename airbot_2d.lua"
        output="screen">
        <remap from="imu" to="/livox/imu" />
    </node>

    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
        type="cartographer_occupancy_grid_node" args="-resolution 0.01" />

</launch>